### PR TITLE
[1pt]: Remove FIM3 and most references to GMS

### PR DESCRIPTION
**Pull request by RobHanna-NOAA**
This update removes the remaining elements of FIM3 code.  It further removes the phrases "GMS" as basically the entire FIM4 model. FIM4 is GMS. With removing FIM3, it also means remove concepts of "MS" and "FR" which were no longer relevant in FIM4.  There are only a few remaining places that will continue with the phrase "GMS" which is in some inundation files which are being re-evaluated.  Some deprecated files have been removed and some subfolders removed.

There are a lot of duplicate explanations for some of the changes, so here is a shortcut system.

desc 1:  Remove or rename values based on phrase "GMS, MS and/or FR"
desc 2:  Moved file from the /src/gms folder to /src  or /tools/gms_tools to /tools
desc 3:  No longer needed as we now use the `fim_pipeline.sh` processing model.

### Removals

- `data`
    - `acquire_and_preprocess_inputs.py`:  No longer needed
- `gms_pipeline.sh` : see desc 3
- `gms_run_branch.sh` : see desc 3
- `gms_run_post_processing.sh` : see desc 3
- `gms_run_unit.sh` : see desc 3
- `src`
    - `gms`
        - `init.py` : folder removed, no longer needed.
        - `aggregate_branch_lists.py`: no longer needed.  Newer version already exists in src directory.
        - `remove_error_branches.py` :  see desc 3
        - `run_by_unit.sh` : see desc 3
        - `test_new_crosswalk.sh` : no longer needed
        - `time_and_tee_run_by_branch.sh` : see desc 3
        - `time_and_tee_run_by_unit.sh` : see desc 3
    - `output_cleanup.py` : see desc 3
 - `tools/gms_tools`
     - `init.py` : folder removed, no longer needed.

### Changes

- `config`
   - `deny_branch_unittests.lst` :  renamed from `deny_gms_branch_unittests.lst`
   - `deny_branch_zero.lst` : renamed from `deny_gms_branch_zero.lst`
   - `deny_branches.lst` :  renamed from `deny_gms_branches.lst`
   - `deny_unit.lst`  : renamed from `deny_gms_unit.lst`
   - `params_template.env` : see desc 1
 
- `data`
    - `nws`
        - `preprocess_ahps_nws.py`:   Added deprecation note: If reused, it needs review and/or upgrades.
    - `acquire_and_preprocess_3dep_dems.py` : see desc 1
 - `fim_post_processing.sh` : see desc 1, plus a small pathing change.
 - `fim_pre_processing.sh` : see desc 1
 - ` src`
     - `add_crosswalk.py` : see desc 1. Also cleaned up some formatting and commented out a code block in favor of a better way to pass args from "__main__"
     - `bash_variables.env` : see desc 1
     - `buffer_stream_branches.py` : see desc 2
     - `clip_rasters_to_branches.py` : see desc 2
     - `crosswalk_nwm_demDerived.py` :  see desc 1 and desc 2
     - `delineate_hydros_and_produce_HAND.sh` : see desc 1 and desc 2
     - `derive_level_paths.py`  :  see desc 1 and desc 2
     - `edit_points.py` : see desc  2
     - `filter_inputs_by_huc.py`: see desc 1 and desc 2
     - `finalize_srcs.py`:  see desc 2
     - `generate_branch_list.py` : see desc 1
     - `make_rem.py` : see desc 2
     - `make_dem.py` : see desc  2
     - `outputs_cleanup.py`:  see desc 1
     - `process_branch.sh`:  see desc 1
     - `query_vectors_by_branch_polygons.py`: see desc 2
     - `reset_mannings.py` : see desc 2
     - `run_by_branch.sh`:  see desc 1
     - `run_unit_wb.sh`: see desc 1
     - `stream_branches.py`:  see desc 2
     - `subset_catch_list_by_branch_id.py`: see desc 2
     - `toDo.md`: see desc 2
     - `usgs_gage_aggregate.py`:  see desc 1
     - `usgs_gage_unit_setup.py` : see desc 1
     - `utils`
         - `fim_enums.py` : see desc 1
         
- `tools`
    - `combine_crosswalk_tables.py` : see desc 2
    - `compare_ms_and_non_ms_metrics.py` : see desc 2
    - `compile_comp_stats.py`: see desc 2  and added note about possible deprecation.
    - `compile_computation_stats.py` : see desc 2  and added note about possible deprecation.
    - `composite_inundation.py` : see desc 1 : note.. references a file called inundate_gms which retains it's name for now.
    - `consolidate_metrics.py`: added note about possible deprecation.
    - `copy_test_case_folders.py`: see desc 1
    - `eval_plots.py` : see desc 1
    - `evaluate_continuity.py`: see desc 2
    - `find_max_catchment_breadth.py` : see desc 2
    - `generate_categorical_fim_mapping.py` : see desc 1
    - `inundate_gms.py`: see desc 1 and desc 2. Note: This file has retained its name with the phrase "gms" in it as it might be upgraded later and there are some similar files with similar names.
    - `inundate_nation.py` : see desc 1
    - `inundation.py`:  text styling change
    - `make_boxes_from_bounds.py`: text styling change
    - `mosaic_inundation.py`:  see desc 1 and desc 2
    - `overlapping_inundation.py`: see desc 2
    - `plots.py` : see desc 2
    - `run_test_case.py`:  see desc 1
    - `synthesize_test_cases.py`: see desc 1
    
- `unit_tests`
    - `README.md`: see desc 1
    - `__template_unittests.py`: see desc 1
    - `check_unit_errors_params.json`  and `check_unit_errors_unittests.py` : see desc 1
    - `derive_level_paths_params.json` and `derive_level_paths_unittests.py` : see desc 1 and desc 2
    - `filter_catchments_and_add_attributes_unittests.py`: see desc 1
    - `outputs_cleanup_params.json` and `outputs_cleanup_unittests.py`: see desc 1 and desc 2
    - `split_flows_unittests.py` : see desc 1
    - `tools`
        - `inundate_gms_params.json` and `inundate_gms_unittests.py`: see desc 1 and desc 2

## Testing

Testing was done in a against a couple of hucs including one that had benchmark data. Then it was run against synthesize_test_case.py and eval_plots.py to validate that those tools continued to run.

An alpha domain run and alpha test is was run and it was fine.

### CODE REVIEW

File: src/gms/aggregate_branch_lists.py [hunk 1]
```diff
@@ -1,57 +0,0 @@
-#!/usr/bin/env python3
```
Comment (robgpita-noaa) [inline src/gms/aggregate_branch_lists.py:1]: This file was deleted, but is being used/imported by `/tools/copy_test_case_folders.py`.

File: src/gms/generate_branch_list.py [hunk 2]
```diff
@@ -1,36 +0,0 @@
-#!/usr/bin/env python3
```
Comment (robgpita-noaa) [inline src/gms/generate_branch_list.py:1]: This file was deleted but being used elsewhere.

File: tools/copy_test_case_folders.py [hunk 3]
```diff
@@ -73,11 +73,11 @@ def copy_folders(folder_name_list,
...
 
-    if create_gms_input_list == True:
+    if create_fim_input_list == True:
         # call this code, which scans each huc (unit) directory looking for the branch_id.lst
-        # and adds them together to create the gms_inputs.csv file
+        # and adds them together to create the fim_inputs.csv file
         # Note: folder_name_list needs to be a huc list to work)
-        agg.aggregate_inputs_for_gms(folder_name_list, target_dir, "gms_inputs.csv")
+        agg.aggregate_branch_lists(folder_name_list, target_dir, "gms_inputs.csv")
         print("gms_inputs.csv created")
```
Comment (robgpita-noaa) [inline tools/copy_test_case_folders.py:81]: "fim_inputs.csv. created"?

File: tools/copy_test_case_folders.py [hunk 4]
```diff
@@ -6,13 +6,13 @@
 import sys
 
 # importing python folders in other direcories
-sys.path.append('/foss_fim/src/gms/')
+sys.path.append('/foss_fim/src/')
 import aggregate_branch_lists as agg
```
Comment (robgpita-noaa) [inline tools/copy_test_case_folders.py:10]: This file doesn't exist in `/tools`

**Review by robgpita-noaa — CHANGES_REQUESTED** [review]
Lots of changed files here, good work putting it all together! Only thing I'd say is that some of the files previously in `/src/gms/`  were removed, but are still being imported by other modules.

File: tools/copy_test_case_folders.py (hunk 4 above)
Comment (RobHanna-NOAA) [reply tools/copy_test_case_folders.py:10]: tools/copy_test_case_folders.py  --- I see it in the git branch via github, so not sure what's up.

File: tools/synthesize_test_cases.py [hunk 5]
```diff
@@ -233,56 +235,40 @@ def progress_bar_handler(executor_dict, verbose, desc):
...
      Notes:
-       - gms_input.csv MUST be in the folder suggested.
+       - fim_input.csv MUST be in the folder suggested.
        - the -v param is the name in the folder in the "outputs/" directory where the test hucs are at.
            It also becomes the folder names inside the test_case folders when done.
        - the -vg param may not be working (will be assessed better on later releases).
        - Find a balance between -jh (number of jobs for hucs) versus -jb (number of jobs for branches)
          on quick tests on a 96 core machine, we tried [1 @ 80], [2 @ 40], and [3 @ 25] (and others).
-       -jb 3 -jh 25 was noticably better. You can likely go more jb cores with better success, just
+       -jb 3 -ji 25 was noticably better. You can likely go more jb cores with better success, just
```
[the remark this answers is not in the document]
Comment (RobHanna-NOAA) [reply tools/synthesize_test_cases.py:247]: For a short time, I changed it from jb to ji, then changed it back to jb a short while ago. Thanks for pointing this out as I missed some text changes. I will check it in shortely.

File: tools/copy_test_case_folders.py [hunk 6]
```diff
@@ -73,11 +73,11 @@ def copy_folders(folder_name_list,
...
     print(f"{str(ctr)} folders have been copied to {target_dir}")
 
-    if create_gms_input_list == True:
+    if create_fim_input_list == True:
         # call this code, which scans each huc (unit) directory looking for the branch_id.lst
-        # and adds them together to create the gms_inputs.csv file
+        # and adds them together to create the fim_inputs.csv file
         # Note: folder_name_list needs to be a huc list to work)
-        agg.aggregate_inputs_for_gms(folder_name_list, target_dir, "gms_inputs.csv")
+        agg.aggregate_branch_lists(folder_name_list, target_dir, "gms_inputs.csv")
```
[the remark this answers is not in the document]
Comment (RobHanna-NOAA) [reply tools/copy_test_case_folders.py:80]: good catch, thanks.

File: src/gms/aggregate_branch_lists.py (hunk 1 above)
Comment (RobHanna-NOAA) [reply src/gms/aggregate_branch_lists.py:1]: replaced by a newer seperate edition that pre-existed in the src directory. Hopefully the pointers are correct now.

File: src/gms/generate_branch_list.py (hunk 2 above)
Comment (RobHanna-NOAA) [reply src/gms/generate_branch_list.py:1]: restored. thank you.

File: src/gms/aggregate_branch_lists.py (hunk 1 above)
Comment (RobHanna-NOAA) [reply src/gms/aggregate_branch_lists.py:1]: added possible deprecation warning

File: tools/copy_test_case_folders.py (hunk 4 above)
Comment (RobHanna-NOAA) [reply tools/copy_test_case_folders.py:10]: added a possible deprecation warning, but fixed up a few things in it anyways.

**Review by robgpita-noaa — APPROVED** [review]

**Review by CarsonPruitt-NOAA — APPROVED** [review]

**Review by CarsonPruitt-NOAA — APPROVED** [review]

**Review by CarsonPruitt-NOAA — APPROVED** [review]

**Pull request merged**